src/content/en/android/custom-tabs/headers/index.md

```diff
@@ -0,0 +1,212 @@
+origin once the link is verified.
+
+### Create custom tab intent with extra headers
+There are multiple ways for creating a [Custom Tabs intent][11]. You can use the builder available
+in androidX by adding the library to the build dependencies:
+
+```gradle
+implementation 'androidx.browser:browser:1.2.0'
+```
+
+Build the intent and add extra headers:
+
+```java
+CustomTabsIntent constructExtraHeadersIntent(CustomTabsSession session) {
+    CustomTabsIntent intent = new CustomTabsIntent.Builder(session).build();
+
+    // Example non-cors-whitelisted headers.
+    Bundle headers = new Bundle();
+    headers.putString("bearer-token", "Some token");
+    headers.putString("redirect-url", "Some redirect url");
+    intent.intent.putExtra(Browser.EXTRA_HEADERS, headers);
+    return intent;
+}
+```
+### Set up a custom tabs connection to validate the asset link
+
+```java
+// Set up a connection that warms up and validates a session.
+CustomTabsServiceConnection connection = new CustomTabsServiceConnection() {
+    @Override
+    public void onCustomTabsServiceConnected(@NonNull ComponentName name, 
+        @NonNull CustomTabsClient client) {
+        // Create session after service connected.
+        mSession = client.newSession(callback);
+        client.warmup(0);
+        // Validate the session as the same origin to allow cross origin headers.
+        mSession.validateRelationship(CustomTabsService.RELATION_USE_AS_ORIGIN, 
+            Uri.parse(url), null);
+    }
+    @Override
+    public void onServiceDisconnected(ComponentName componentName) { }
+};
+```
+
+### Set up a callback that launches the intent after validation
+```java
+// Set up a callback that launches the intent after session validated.
+CustomTabsCallback callback = new CustomTabsCallback() {
+    @Override
+    public void onRelationshipValidationResult(int relation, @NonNull Uri requestedOrigin, 
+        boolean result, @Nullable Bundle extras) {
+        // Launch custom tabs intent after session was validated as the same origin.
+        CustomTabsIntent intent = constructExtraHeadersIntent(mSession);
+        intent.launchUrl(MainActivity.this, Uri.parse(url));
+    }
+};
+```
+
+### Bind the custom tabs service connection
+
+Binding the service launches the service and the connection’s `onCustomTabsServiceConnected()`
+will be called eventually. Don’t forget to unbind the service appropriately. Binding and unbinding
+is commonly done in the `onStart()` and `onStop()` activity lifecycle methods.
+
+```java
+// Bind the custom tabs service connection.
+// Call this in onStart()
+CustomTabsClient.bindCustomTabsService(this,
+    CustomTabsClient.getPackageName(MainActivity.this, null), connection);
+
+// …
+// Unbind the custom tabs service.
+// Call this in onStop().
+unbindService(connection);
+```
+
+### Demo application code
+
+More details about Custom Tabs Service can be found [here][12]. A working example app can be found
```

Comment:
```suggestion
You can find more details about Custom Tabs Service [here][12]. A working example app can be found
```

Reply:
Constructions like 'can be found' and 'will be done' are not considered the best possible writing in English. Replace them with phrases like 'You can find' and 'the system will'. I won't hold up the article, but if you feel inclined, it would improve the writing.